### DISCUSSION TITLE
python type for `std::array`

### DISCUSSION THREAD

**Discussion by vnghia** [wjakob/nanobind#50] [Q&A] [answered]
I think it is good if we could add a caster for `std::array`, we already have `std::vector` so I think it won't be too hard. For my use case, I have some `constexpr std::array` that I want to expose to python without copying it to `std::vector` beforehand. I can send a PR if you think this is a good idea. Thank you very much.

**Comment by vnghia** [comment, marked as answer]
It will be just as simple as:
```c++
template <typename Type, size_t N>
struct nb::detail::type_caster<std::array<Type, N>>
    : nb::detail::list_caster<std::array<Type, N>, Type> {};
```

**Reply by wjakob** [reply]
Could you submit a PR with tests? In that case, I would merge it. Thanks!

**Reply by vnghia** [reply]
I realized that we can not cast from python list to `std::array` because the size has to be known at compile time. I am thinking of overriding the `from_python` of `list_caster` and return false or writing another caster entirely. What do you think ?

**Reply by vnghia** [reply]
or we could change this line:
https://github.com/wjakob/nanobind/blob/d9d0380554f52bbcbf7eb3aba99ac19d7a4131e8/include/nanobind/stl/detail/nb_list.h#L37
to something like `value[i] = ...` and discard the rest if the list is longer or leave unintialized if the list is shoter than N.

**Reply by wjakob** [reply]
(Marking as answered since a type caster was added in the meantime)
